go/inst/query_string_provider.go

```diff
@@ -0,0 +1,351 @@
+	master_host:               "Source_Host",
+	master_port:               "Source_Port",
+	seconds_behind_master:     "Seconds_Behind_Source",
+	master_ssl_allowed:        "Source_SSL_Allowed",
+	show_slave_hosts:          "show replicas",
+
+	stop_slave_io_thread:                          "stop replica io_thread",
+	stop_slave_sql_thread:                         "stop replica sql_thread",
+	start_slave_sql_thread:                        "start replica sql_thread",
+	start_slave_io_thread:                         "start replica io_thread",
+	stop_slave:                                    "stop replica",
+	start_slave_until_master_log:                  "start replica until source_log_file=?, source_log_pos=?",
+	master_user_param:                             "source_user = ?",
+	master_password_param:                         "source_password = ?",
+	master_ssl_ca_param:                           "source_ssl_ca = ?",
+	master_ssl_cert_param:                         "source_ssl_cert = ?",
+	master_ssl:                                    "source_ssl",
+	master_ssl_key_param:                          "source_ssl_key = ?",
+	change_master_to_master_ssl:                   "change replication source to source_ssl=1",
+	reset_slave:                                   "reset replica",
+	change_master_to_master_host_port:             "change replication source to source_host=?, source_port=?",
+	change_master_to_master_host_port_log_gtid_no: "change replication source to source_host=?, source_port=?, source_log_file=?, source_log_pos=?, source_use_gtid=no",
+	change_master_to_master_host_port_log_autoposition_no: "change replication source to source_host=?, source_port=?, source_log_file=?, source_log_pos=?, source_auto_position=0",
+	change_master_to_master_host_port_autoposition_yes:    "change replication source to source_host=?, source_port=?, source_auto_position=1",
+	change_master_to_master_host_port_log:                 "change replication source to source_host=?, source_port=?, source_log_file=?, source_log_pos=?",
+	change_master_to_master_host:                          "change replication source to source_host='_'",
+	reset_slave_50603_all:                                 "reset replica /*!50603 all */",
+	reset_master:                                          "reset binary logs and gtids",
+	select_master_pos_wait:                                "select source_pos_wait(?, ?)",
+	change_master_to_with_params:                          "change replication source to %s",
+	change_master_to_master_delay:                         "change replication source to source_delay=%d",
+	set_sql_slave_skip_counter:                            "set global sql_replica_skip_counter := 1",
+}
+
+var queryStringProvider80 = QueryStringProvider{
+	queries: queryStrings80,
+}
+
+var queryStringProvider84 = QueryStringProvider{
+	queries: queryStrings84,
+}
+
+func GetQueryStringProvider(version string) QueryStringProvider {
+	if strings.HasPrefix(version, "8.4") {
```

Comment:
If not now, in future we will need to normalize version numbers as intergers for comparisions

Reply:
Let's keep it simple for now. We can always do more magic here when needed.